app/modules/analysis/topic_analysis.py

```
@@ -31,3 +31,196 @@
+    """Load translated topics/words from JSON."""
+    with open(data_path, "r", encoding="utf-8") as f:
+        return json.load(f)
+
+
+# -----------------------------
+#  Compute distances per topic
+# -----------------------------
+def compute_topic_distances(translated_data: List[Dict]) -> Dict:
+    """
+    For each topic:
+      – detect languages dynamically
```

Ruff (0.14.6)

Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)